script.js

```diff
@@ -1,4 +1,214 @@
+//Base Features
+//P1: Create a set of cards into an array
+var makeDeck = function () {
+  // Initialise an empty deck array
+  var cardDeck = []; // Initialise an array of the 4 suits in our deck. We will loop over this array.
+  var suits = ["Hearts", "Diamonds", "Clubs", "Spades"]; // Loop over the suits array
+
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex]; // Loop from 1 to 13 to create all cards for a given suit // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12. // This is an example of a loop without an array.
+
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter; // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+
+      if (cardName == 1) {
+        cardName = "Ace";
+      } else if (cardName == 11) {
+        cardName = "Jack";
+      } else if (cardName == 12) {
+        cardName = "Queen";
+      } else if (cardName == 13) {
+        cardName = "King";
+      }
+      // Create a new card with the current name, suit, and rank
+      //P1: Cards are objects containing (name,suit,rank)
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      }; // Add the new card to the deck
+
+      cardDeck.push(card); // Increment rankCounter to iterate over the next rank
+
+      rankCounter += 1;
+    } // Increment the suit index to iterate over the next suit
+
+    suitIndex += 1;
+  } // Return the completed card deck
+
+  return cardDeck;
+};
+
+//P1: Create a Function for shuffling the deck.
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length); // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex]; // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex]; // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard; // Increment currentIndex to shuffle the next pair of cards
+    currentIndex += 1;
+  } // Return the shuffled deck
+  return cardDeck;
+};
+var deck = shuffleCards(makeDeck());
+
+//Global Variables
+//Default Mode is player turn
+var currentPlayer = "playerTurn";
+//Player & Computer cards in an array
+playerCards = [];
+computerCards = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  currentPlayer = "playerTurn";
+  var deck = shuffleCards(makeDeck());
```

Comment:
why did you declared var deck = shuffleCards(makeDeck()); twice ?

Reply:
Not needed I have redone the whole code as it doesn't emphasises functions to be able to refactor easily